app/assets/javascripts/discourse/app/lib/form-template-validation.js

```diff
@@ -20,9 +24,11 @@ export default function prepareFormTemplateData(form, formTemplate) {
   const formData = new FormData(form);
 
   // Validate the form template
-  _validateFormTemplateData(form);
-  if (!form.checkValidity()) {
-    return false;
+  if (checkFormValidity) {
```

Comment:
why do we sometimes need to not check validity anymore ?

Reply:
This method runs when forms are cooked but we don't want to check validity yet (as the user is typing), otherwise it won't cook and show all the validation issues to the user right away